debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/KafkaConnectRecordParser.java

```diff
@@ -192,15 +194,18 @@ protected void parseValueFields(Struct value, Record r) {
         // TODO: error handle before is NULL
         Struct before = value.getStruct("before");
         
-        // in case of delete, the before struct contains the values required in cases like DELETE-INSERT conflicts(unique key columns)
-        if (r.op.equals("d") && before != null) {
-            after = before;
+
+        if (after == null && !r.op.equals("d")) {
+            throw new RuntimeException(String.format("after struct is null for record with op=%s", r.op));
         }
-        if (after == null) {
-            return;
+        if (before == null && !r.op.equals("c")){
+            throw new RuntimeException(String.format("before struct is null for record with op=%s", r.op));
         }
-
-        for (Field f : after.schema().fields()) {
+        Struct beforeOrAfter = after;
+        if (r.op.equals("d")){
+            beforeOrAfter = before;
+        }
+        for (Field f : beforeOrAfter.schema().fields()) {
```

Comment:
Nit: you can just store fields out from `after` and `before` struct instead of complete structs in `beforeOrAfter` var.